core/src/main/scala/spinal/core/internals/ComponentEmitterVerilog.scala

```diff
@@ -494,10 +494,18 @@ class ComponentEmitterVerilog(
                 case `ERROR` => "ERROR"
                 case `FAILURE` => "FAILURE"
               }
-              b ++= s"${tab}if(!$cond) begin\n"
-              b ++= s"""${tab}  $$display("$severity $frontString"$backString);\n"""
-              if (assertStatement.severity == `FAILURE`) b ++= tab + "  $finish;\n"
-              b ++= s"${tab}end\n"
+              b ++= s"${tab}`ifdef FORMAL\n"
```

Comment:
Why not using similar things as https://github.com/SpinalHDL/SpinalHDL/blob/fbc7dc94dc14b0fb367d1f326fba14016f0f50c2/core/src/main/scala/spinal/core/internals/ComponentEmitterVerilog.scala#L516 ?

Reply:
Because if I understand it correctly, those formal statements are not part of Verilog, but are a Yosys-specific extension. I can still add the `if` as below if you prefer, but the generated code *must* contain an `ifdef FORMAL` or otherwise it won't be compliant.